### PR TITLE
BE-11-add-quizzes

### PR DESCRIPTION
**Pull request by Randayy**

### CODE REVIEW

File: app/repositories/quiz_repository.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                status_code=403, detail='You are not owner or admin of company with this quiz question, you are not permitted to see this)')
+
+    async def create_quiz(self, quiz_data: dict, company_id: UUID):
+        new_quiz = Quiz(
+            title=quiz_data['title'],
+            description=quiz_data.get('description'),
+            frequency_days=quiz_data['frequency_days'],
+            company_id=company_id
+        )
+        self.db.add(new_quiz)
```
Comment (Artemooon) [inline app/repositories/quiz_repository.py:113]: use add_all()

File: app/repositories/quiz_repository.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            )
+            self.db.add(question)
+
+            for answer_data in question_data['answers']:
+                answer = Answer(
+                    answer_text=answer_data['answer_text'],
+                    is_correct=answer_data['is_correct'],
+                    question=question
+                )
+                self.db.add(answer)
```
Comment (Artemooon) [inline app/repositories/quiz_repository.py]: use add_all

File: app/repositories/quiz_repository.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            raise HTTPException(
+                status_code=403, detail='You are not member of company with this quiz, you are not permitted to see this)')
+
+        if company_member.role != Role.ADMIN and company_member.role != Role.OWNER:
+            raise HTTPException(
+                status_code=403, detail='You are not owner or admin of company with this quiz, you are not permitted to see this)')
+
+    async def check_if_user_is_owner_or_admin_in_this_quiz_company(self, current_user_id: UUID, question_id: UUID):
+        question_select = select(Question).options(
+            joinedload(Question.quiz)).where(Question.id == question_id)
```
Comment (Artemooon) [inline app/repositories/quiz_repository.py]: why do you use joinedload here?

File: app/routers/quiz_routes.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@quiz_router.delete("/quizzes/{company_id}/delete/{quiz_id}")
+async def delete_quiz(company_id: UUID, quiz_id: UUID, db: AsyncSession = Depends(get_session), current_user: User = Depends(get_current_user_from_token)):
+    service = QuizService(db)
+    result = await service.delete_quiz(quiz_id, current_user, company_id)
+    return result
+
+
+@quiz_router.patch("/quizzes/{company_id}/update/{quiz_id}")
```
Comment (Artemooon) [inline app/routers/quiz_routes.py]: you do not need to pass company_id here, as quiz table stores it
do just: /quizzes/{quiz_id}/ and take company_id from the table

File: app/routers/quiz_routes.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@quiz_router.get("/questions/{question_id}/answers")
+async def get_question_answers(question_id: UUID, db: AsyncSession = Depends(get_session), current_user: User = Depends(get_current_user_from_token)):
+    service = QuizService(db)
+    result = await service.get_question_answers_to_response(question_id, current_user)
+    return result
+
+
+@quiz_router.delete("/quizzes/{company_id}/delete/{quiz_id}")
```
Comment (Artemooon) [inline app/routers/quiz_routes.py]: here the same

File: app/routers/quiz_routes.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+from app.db.user_models import User, Company
+from app.schemas.quiz_shemas import QuizCreateSchema, QuizUpdateSchema
+from app.services.quiz_service import QuizService
+from app.services.user_service import get_current_user_from_token
+from uuid import UUID
+
+quiz_router = APIRouter(tags=["Quizzes"])
+
+
+@quiz_router.post("/quizzes/{company_id}/create")
```
Comment (Artemooon) [inline app/routers/quiz_routes.py]: add response model for every router

File: app/repositories/quiz_repository.py (hunk 1 above)
Comment (Randayy) [reply app/repositories/quiz_repository.py:113]: used

Comment (Randayy) [reply app/repositories/quiz_repository.py:113]: also found this in internet

File: app/repositories/quiz_repository.py (hunk 2 above)
Comment (Randayy) [reply app/repositories/quiz_repository.py]: fixed

File: app/routers/quiz_routes.py (hunk 4 above)
Comment (Randayy) [reply app/routers/quiz_routes.py]: fixed

File: app/routers/quiz_routes.py (hunk 5 above)
Comment (Randayy) [reply app/routers/quiz_routes.py]: fixed

File: app/routers/quiz_routes.py (hunk 6 above)
Comment (Randayy) [reply app/routers/quiz_routes.py]: added

File: app/repositories/quiz_repository.py (hunk 3 above)
Comment (Randayy) [reply app/repositories/quiz_repository.py]: deleted this , I have some errors when creating this and next step I need to take question.quiz I thought that I can with join get it faster , but I don't need to use any join here

**Review by Artemooon — APPROVED** [review]

**Pull request merged**